frontend/src/entities/group/create/group-create.tsx

```diff
@@ -0,0 +1,217 @@
+import { useAppDispatch } from '../../../config/store';
+import NotificationService from '../../../shared/layout/notification/notification.service';
+import { useNavigate, useParams } from 'react-router-dom';
+import { Button, Container, FormField, Header, Input, SpaceBetween, Textarea, } from '@cloudscape-design/components';
+import Form from '@cloudscape-design/components/form';
+import ContentLayout from '../../../shared/layout/content-layout';
+import { createGroup, getGroup, updateGroup } from '../group.reducer';
+
+export type GroupCreateProperties = {
+    isEdit?: boolean;
+};
+
+export function GroupCreate ({isEdit}: GroupCreateProperties) {
+    const dispatch = useAppDispatch();
+    const notificationService = NotificationService(dispatch);
+    const navigate = useNavigate();
+    const {groupName} = useParams();
+    const [initialLoaded, setInitialLoaded] = useState(false);
+
+    const groupSchema = z.object({
+        name: z
+            .string()
+            .min(3, {
+                message: 'Group name must be longer than 3 characters',
+            })
+            .max(24, {
+                message: 'Group name cannot be more than 24 characters',
+            })
+            .regex(/^[a-zA-Z0-9]+$/, {
+                message: 'Group name can only contain alphanumeric characters.',
+            }),
+        description: z
+            .string({
+                invalid_type_error: 'Group description must be a string',
+            })
+            .min(1, {
+                message: 'Group description cannot be empty',
+            })
+            .max(4000, {
+                message: 'Group description cannot be more than 4000 characters.',
+            })
+            .regex(/[ -~]/, {
+                message: 'Group description can only contain printable characters',
+            }),
+    });
+
+    const {state, setState, setFields, touchFields} = useValidationReducer(groupSchema, {
+        validateAll: false,
+        needsValidation: false,
+        form: {
+            name: '',
+            description: '',
+        },
+        touched: {},
+        formErrors: {} as any,
+        formSubmitting: false,
+    });
+
+    useMemo(() => {
+        if (isEdit && !initialLoaded) {
+            setState({formSubmitting: true});
+            dispatch(getGroup(groupName)).then((response) => {
+                if (response.payload) {
+                    setInitialLoaded(true);
+                    setState({
+                        form: {
+                            name: response.payload?.data.group?.name || '',
+                            description: response.payload?.data.group?.description || '',
+                        },
+                        formSubmitting: false
+                    });
+                } else {
+                    navigate('/404');
+                }
+            });
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Do these exhaustive checks have to be disabled everywhere or is that just from copying and pasting?

Reply:
this particular one was needed due to setState causing this to continuously reload if it was in the dependency list. I was able to remove the others in `group-detail.tsx` though.